src/main/java/com/k3project/demo/security/CustomUserDetailsService.java

```diff
@@ -0,0 +1,46 @@
+package com.k3project.demo.security;
+
+import com.k3project.demo.entity.Role;
+import com.k3project.demo.entity.UserEntity;
+import com.k3project.demo.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    @Autowired
+    public CustomUserDetailsService(UserRepository userRepository) {
+        this.userRepository = userRepository;
+    }
+    @Override
+    public UserDetails loadUserByUsername(String firstName) throws UsernameNotFoundException {
+        System.out.println("tusom++++++++++++");
```

Comment:
Delete all unnecessary prints and use at least one log in method and format the code more readable. Use on class level (@Slf4j)

Reply:
done